sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelHealthChecker.java

```diff
@@ -380,7 +380,7 @@ private String idleConnectionValidation(Timestamps timestamps, Instant currentTi
         String errorMessage = StringUtils.EMPTY;
 
         if (this.idleConnectionTimeoutInNanos > 0L) {
-            if (Duration.between(currentTime, timestamps.lastChannelReadTime()).toNanos() > this.idleConnectionTimeoutInNanos) {
+            if (Duration.between(timestamps.lastChannelReadTime(), currentTime).toNanos() > this.idleConnectionTimeoutInNanos) {
```

Comment:
Good catch :)
@xinlian12  was there a reason you added `currentTime `as first parameter?

Reply:
good catch, no, it is bug. I think I was meant to change to use Duration.between, but forgot to change the sequence.